### PR TITLE
Confucianism, Sramanic, Bhakti society + minor fixes

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

**Comment by IhateTrains** [comment]
Should all my previous commits be listed?

**Comment by loup99** [comment]
Usually they aren't, but as I'm the owner of the repository it looks different for me.

File: WTWSMS/common/religions/00_religions.txt [hunk 1]
```diff
@@ -1960,8 +1958,6 @@ pagan_group = {
 		
 		religious_clothing_head = 3
 		religious_clothing_priest = 3
-
-		allow_in_ruler_designer = no
```
Comment (loup99) [inline WTWSMS/common/religions/00_religions.txt]: Why did you remove this?

File: WTWSMS/common/religions/00_religions.txt [hunk 2]
```diff
@@ -4337,7 +4330,6 @@ pagan_group = {
 		max_consorts = 3
 		
 		ai_convert_same_group = 0 # never
-		ai_convert_other_group = 1 # only when zealous
```
Comment (loup99) [inline WTWSMS/common/religions/00_religions.txt]: Why did you remove this?

File: WTWSMS/common/traits/02_traits.txt [hunk 3]
```diff
@@ -2555,7 +2555,6 @@ lovers_pox = {
 	sex_appeal_opinion = -5
 	
 	customizer = no
-	customizer = no
```
Comment (loup99) [inline WTWSMS/common/traits/02_traits.txt]: For reference, this is a vanilla trait, which is why I haven't touched it.

File: WTWSMS/history/provinces/100 - Avranches.txt [hunk 4]
```diff
@@ -22,8 +22,6 @@ religion = celtic_pagan
 terrain = forest
 # History
 1.1.1 = { # Viking takeover
-	remove_settlement = b_caen
-	remove_settlement = b_bayeux
```
Comment (loup99) [inline WTWSMS/history/provinces/100 - Avranches.txt]: I'm not sure what is supposed to be in here.

File: WTWSMS/common/religions/00_religions.txt (hunk 1 above)
Comment (IhateTrains) [inline WTWSMS/common/religions/00_religions.txt]: It was duplicated

File: WTWSMS/common/religions/00_religions.txt (hunk 2 above)
Comment (IhateTrains) [inline WTWSMS/common/religions/00_religions.txt]: Also duplicated

File: WTWSMS/history/provinces/100 - Avranches.txt (hunk 4 above)
Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: These settlements weren't built, so there's no point in removing them

File: WTWSMS/common/religions/00_religions.txt (hunk 1 above)
Comment (IhateTrains) [inline WTWSMS/common/religions/00_religions.txt]: Look a few lines above

Comment (loup99) [inline WTWSMS/common/religions/00_religions.txt]: Okay, I missed it.

File: WTWSMS/common/religions/00_religions.txt (hunk 2 above)
Comment (loup99) [inline WTWSMS/common/religions/00_religions.txt]: It should be `ai_convert_other_group = 1 # only when zealous` rather than `ai_convert_other_group = 2 # always try to convert`.

File: WTWSMS/history/provinces/100 - Avranches.txt (hunk 4 above)
Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: Maybe someone forgot to change the date of this event

Comment (loup99) [inline WTWSMS/history/provinces/100 - Avranches.txt]: Yes, but my statement was that I don't get the block itself at all, given that the viking takeover itself is anachronistic at that date.

File: WTWSMS/common/religions/00_religions.txt (hunk 2 above)
Comment (IhateTrains) [inline WTWSMS/common/religions/00_religions.txt]: OK, I deleted 1 because Cybelians had 2. Could you change it? I'm afraid I won't be able to push any commits today.

File: WTWSMS/history/provinces/100 - Avranches.txt (hunk 4 above)
Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: I've found this in Wikipedia:
The regions that later became the Duchies of Normandy and Brittany each experienced devastating Viking raids, with Brittany occupied by Vikings from 907 to 937. In 933 Avranches and its territory, the Avranchin, were ceded to the Normans.

File: WTWSMS/common/religions/00_religions.txt (hunk 2 above)
Comment (loup99) [inline WTWSMS/common/religions/00_religions.txt]: OK, I will change it once I merge.

File: WTWSMS/history/provinces/100 - Avranches.txt (hunk 4 above)
Comment (loup99) [inline WTWSMS/history/provinces/100 - Avranches.txt]: Yes, that I already know, but I don't get the date.

Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: Are you going to unlock vanilla bookmarks in future releases?

Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: It's an obvious error, we should just change the date and let these two lines stay :)

Comment (loup99) [inline WTWSMS/history/provinces/100 - Avranches.txt]: No, they have been removed and won't return. The mod is an overhaul, there is no need for us to support vanilla bookmarks which would be inaccurate. Those who want to experience those bookmarks can play directly on the vanilla game. 😉

**Comment by IhateTrains** [comment]
Yes, I try to be up-to-date. Maybe merging from command line will prevent the older commits from being listed.

File: WTWSMS/history/provinces/100 - Avranches.txt (hunk 4 above)
Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: OK, I will probably fix the province history anyway to clean Validator output

Comment (IhateTrains) [inline WTWSMS/history/provinces/100 - Avranches.txt]: Apropos bookmarks, are 395 and 6xx still planned?

Comment (loup99) [inline WTWSMS/history/provinces/100 - Avranches.txt]: It could also be that this is intended but badly explained. Sometimes the comment aren't always indicative as it is copy/paste from vanilla.

On bookmarks, have you been following the mod for a long time? Because our plans haven't changed, it is just that we need to finalise the goals for the full release before expanding history.

File: WTWSMS/common/religions/00_religions.txt (hunk 2 above)
Comment (IhateTrains) [inline WTWSMS/common/religions/00_religions.txt]: I'm fixing it myself in the next commit.

**Comment by loup99** [comment]
All post-800 or so history and characters should actually be removed, but it is on hold due to potential conflicts.

**Comment by IhateTrains** [comment]
--- Error 1 of 1 ---
At <mod>\history\provinces\1225 - Puri.txt [religion] (Line 19, column 1):
"sarnaist" is not a valid Religion.

what religion might it be?

**Comment by IhateTrains** [comment]
Have you found the name ofHave you found the name of the loadscreen effect? the loadscreen effect?

**Comment by IhateTrains** [comment]
Check out the western Imperial Regalia icons I added in the last commit. They deserve a more unique icons in the future.

**Comment by loup99** [comment]
For some reason I no longer get alerts for this, sorry for my lack of replies due to that. Sarnaist is a religion @EmperorEnlil has coded but not added. It should be comment out and replaced with the prior defined religion in the file.

I haven't found the effect.

I love the art (did you do it yourself? Then I want to sign the contract immediately, you would be the perfect new artist 😉 )! However, would it be possible to add more to it or focus on the aspects? The imperial regalia was a series of objects: https://history.stackexchange.com/questions/20684/what-were-the-insignia-of-the-western-roman-empire-and-what-happened-to-them (the link is to a forum where it is discussed, for inspiration, of course not everything needs to be in). Otherwise that can serve as a placeholder, as what you did is already good progress.

**Comment by IhateTrains** [comment]
OK, I will change the religion.

Damn, how about adding an unmodifed version of the mosaic?

This is just a purple version of vanilla Blessed Mantle. Thanks for the link, it will be very useful. I'm planning to add a crown and a lance.

I've got a new question: is it inteded that ERE vassals are feudal?

**Comment by loup99** [comment]
- Yes, but leave it with a hashtag, just so that I don't forget 😛 
- It doesn't fit in with the rest in my opinion to have a mosaic as a loadingscreen. That said the mosaic could come to use elsewhere. This is just my personal subjective opinion though, so I'm open to getting convinced that I'm wrong or reconsider it.
- Okay, but it is well done. If you do know how to dabble with art you are more than welcome!
- Yes, that is intended. I may be able to dig up a link with an explanation on our sub-forum

**Comment by loup99** [comment]
What I did dig up was an idea on adding High Tribal and Sub-Roman government types. But that isn't what I searched for.

**Comment by IhateTrains** [comment]
On 
> Zoroastrian vassals of Persia are feudal government rather than Eransharhr 

Are all vassals supposed to be Eranshahrh? There are jewish feudal, zoroastrian feudal and some tribal.

**Comment by loup99** [comment]
Here is an explanation on that: https://forum.paradoxplaza.com/forum/index.php?threads/testing.961734/page-9#post-22856317

**Comment by loup99** [comment]
Ah, you found the spreadsheet? I must admit the link wasn't the most visible in the group.

The issue with the Eranshahrh government is that it is meant to be the one of those who are either vassals of Persia, the ruler of Persia or of a Parthian or Persian culture and are of a Mazdan religion. So no Christian or Jewish vassal should be Eranshahrh, but all Zoroastrian vassals should.

**Comment by IhateTrains** [comment]
This is my Imperial Regalia idea
plan

**Comment by IhateTrains** [comment]
Please merge, it's becoming a blob :D

**Comment by loup99** [comment]
Okay, I will look through it all and then merge.

File: WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	}
+    
+    weight = {
+		value = 0
+		additive_modifier = {
+			value = 100
+			primary_title = {
+				title = k_lombardia
+			}
+			NOT = { year = 1066 }
```
Comment (loup99) [inline WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt]: Does this need to be in?

File: WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt [hunk 6]
```diff
@@ -0,0 +1,57 @@
+####################
+#### QUALITY 5  ####
+####################
+
+iron_crown_of_lombardy = {
```
Comment (loup99) [inline WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt]: Isn't it better to directly modify the vanilla file?

File: WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt (hunk 5 above)
Comment (IhateTrains) [inline WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt]: `NOT = { year = 1066 }` can be removed

File: WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt (hunk 6 above)
Comment (IhateTrains) [inline WTWSMS/common/artifact_spawns/WtWSMS_artifact_spawns.txt]: ok

**Comment by IhateTrains** [comment]
Good news: I've found the artistic filter! It's called Cut Out. I've prepared two loadscreens using it.

**Comment by loup99** [comment]
Then this one is ready to merge. I'd like to write changelog entries for it though, but I can do that in the merge. 

As for your idea, sounds good as a start! This was what was suggested:
> However perhaps the ERE emperor could receive the imperial regalia as artifacts that give him a nice boost as sole remaining Roman emperor, while the ERE emperor can officially grant Odoacer a Roman title which gives him legitimacy and some form of trait or bonus.
I thought that the emperor could recognise Odoacer in exchange, giving him prestige and a boost (like we sid for the ERE/WRE restoration).

**Comment by loup99** [comment]
Just to explain myself on the changelog, what I usually list is:
- All new stuff we add or balance changes
- All suggestions from the forums, GitHub or group which have been implemented
- All bugs from the forums, GitHub or group which have been fixed
Minor code changes, log or Validator fixes do not need to be listed, unless it is a major issue.

**Comment by loup99** [comment]
The artistic filter is perfect. Could the mosaic be cut so that the ribbons above and below the picture are removed? 

Other than that, I think we can discuss the artifact more elsewhere, so the pull request can be merged! 😃

**Comment by IhateTrains** [comment]
The ribbons aren't visible while loading, they are hidden behind the grey bars with loading and tips.

**Comment by loup99** [comment]
Ah, I had forgotten that while reviewing them. Excellent then, this can be merged.
